packages/descendants/src/utils/findDOMIndex.ts

```diff
@@ -0,0 +1,51 @@
+import { DescendantsList } from '../Descendants.types';
+
+import { isElementPreceding } from './isElementPreceding';
+
+/**
+ * Returns the 1 + the index of the registered descendant immediately preceding the search element.
```

Comment:
Q for tomorrow: I understand that this util is returning the index for the unregistered element, but I'm not sure i understand how this is doing that.

Reply:
Updated the comment to be a bit clearer. Essentially we  crawl the DOM to find the index of the element that's right before our search `element`, then return the index _after_ that (i.e. the element after the element before the search element, is the search element)